pyqldb/session/qldb_session.py

```diff
@@ -100,128 +91,66 @@ def session_token(self):
+        :raises ExecuteError: Error containing the context of a failure during execute.
         """
-
-        while True:
-            transaction = None
-            error = None
-            try:
-                transaction = self._start_transaction()
-                result = query_lambda(Executor(transaction))
-                if isinstance(result, StreamCursor):
-                    # If someone accidentally returned a StreamCursor object which would become invalidated by the
-                    # commit, automatically buffer it to allow them to use the result anyway.
-                    result = BufferedCursor(result)
-                transaction._commit()
-                return result
-            except LambdaAbortedError as lae:
-                self._no_throw_abort(transaction)
-                raise lae
-            except StartTransactionError as ste:
-                error = ste.error
-                self._no_throw_abort(transaction)
-                if context.execution_attempt >= retry_config.retry_limit:
-                    # raise wrapped Error
-                    raise ste.error
-            except RETRYABLE_HTTP_ERRORS as rhe:
-                error = rhe
-                self._no_throw_abort(transaction)
-                if context.execution_attempt >= retry_config.retry_limit:
-                    raise rhe
-                logger.warning('Retryable HTTP error occurred: {}, retrying transaction'.format(rhe))
-            except ClientError as ce:
-                error = ce
-                if is_invalid_session_exception(ce):
-                    self._is_closed = True
-                    raise ce
-                if not is_occ_conflict_exception(ce):
-                    self._no_throw_abort(transaction)
-                if is_occ_conflict_exception(ce) or is_retriable_exception(ce):
-                    if context.execution_attempt >= retry_config.retry_limit:
-                        raise ce
-                    logger.warning('OCC conflict or retriable exception occurred: {}'.format(ce))
-                else:
-                    raise ce
-            except Exception as e:
+        transaction = None
+        transaction_id = None
+        try:
+            transaction = self._start_transaction()
+            result = query_lambda(Executor(transaction))
+            if isinstance(result, StreamCursor):
+                # If someone accidentally returned a StreamCursor object which would become invalidated by the
+                # commit, automatically buffer it to allow them to use the result anyway.
+                result = BufferedCursor(result)
+            transaction._commit()
+            return result
+        except Exception as e:
+            is_retryable = is_retriable_exception(e)
+            is_session_invalid = is_invalid_session_exception(e)
+
+            if is_session_invalid is True and not is_transaction_expired_exception(e):
```

Comment:
Do you need to have `is_session_invalid is True` or could you simply have `is_session_invalid`?

Reply:
We can have simply `is_session_invalid`. Updated here and other occurances